docs/src/user_docs/assembly/code_organization.md

```diff
@@ -3,35 +3,28 @@ A Miden assembly program is just a sequence of instructions each describing a sp
 
 In turn, Miden assembly instructions are just keywords which can be parameterized by zero or more parameters. The notation for specifying parameters is *keyword.param1.param2* - i.e., the parameters are separated by periods. For example, `push.123` instruction denotes a `push` operation which is parameterized by value `123`.
 
-Currently, Miden assembly provides two types of code organization blocks: *programs* and *procedures*. In the future, additional blocks such as modules and functions will be added.
+Miden assembly programs are organized into procedures. Procedures, in turn, can be grouped into modules.
 
-### Programs
-A *program* block is used to define an executable program. A program must start with a `begin` instruction and terminate with an `end` instruction. For example:
-```
-begin
-    <instructions>
-end
-```
-When Miden assembly code is executed, the execution starts at the first instruction following the `begin` instruction of the program. A program block is expected to be the last block in an assembly file and can be followed only by comment blocks.
+Currently, Miden assembly provides two types of code organization blocks: *programs* and *procedures*. In the future, additional blocks such as modules and functions will be added.
```

Comment:
I find this and the above section to be a bit confusing together, especially since modules are discussed below. What's the difference between the future modules and the ones described below? Also, programs contain procedures, not the other way around, so "Miden assembly programs are organized into procedures" is confusing.

I think the new sentence here and the new sentence above should be combined into a paragraph and clarified.